.github/workflows/build.yaml

```diff
@@ -1,15 +1,17 @@
 name: Build
 
 on:
-  push: 
```

Comment:
why did you change the triggers? what's the benefit of the new config?

Reply:
The `pull_request_target` does not run the pipeline on the pull request head commit, it runs on the commit of the base (target) branch - https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target 

This is the reason why the pipeline run on incorrect commit in previous pull request.